src/elements/a-gltf-entity.js

```diff
@@ -0,0 +1,190 @@
+  el.setAttribute("position", {
+    x: node.position.x,
+    y: node.position.y,
+    z: node.position.z
+  });
+  el.setAttribute("rotation", {
+    x: node.rotation.x * THREE.Math.RAD2DEG,
+    y: node.rotation.y * THREE.Math.RAD2DEG,
+    z: node.rotation.z * THREE.Math.RAD2DEG
+  });
+  el.setAttribute("scale", {
+    x: node.scale.x,
+    y: node.scale.y,
+    z: node.scale.z
+  });
+
+  node.position.set(0, 0, 0);
+  node.rotation.set(0, 0, 0);
+  node.scale.set(1, 1, 1);
+
+  el.setObject3D(node.type.toLowerCase(), node);
+
+  children.forEach(childNode => {
+    inflateEntities(classPrefix, el, childNode);
+  });
+};
+
+function attachTemplate(templateEl) {
+  const selector = templateEl.getAttribute("data-selector");
+  const targetEls = templateEl.parentNode.querySelectorAll(selector);
+  const clone = document.importNode(templateEl.content, true);
+  const templateRoot = clone.firstElementChild;
+  const templateRootAttrs = templateRoot.attributes;
+
+  for (var i = 0; i < targetEls.length; i++) {
+    const targetEl = targetEls[i];
+
+    // Merge root element attributes with the target element
+    for (var i = 0; i < templateRootAttrs.length; i++) {
+      targetEl.setAttribute(templateRootAttrs[i].name, templateRootAttrs[i].value);
+    }
+
+    // Append all child elements
+    for (var i = 0; i < templateRoot.children.length; i++) {
+      targetEl.appendChild(document.importNode(templateRoot.children[i], true));
+    }
+  }
+}
+
+AFRAME.registerElement("a-gltf-entity", {
+  prototype: Object.create(AFRAME.AEntity.prototype, {
+    load: {
+      value() {
+        if (this.hasLoaded || !this.parentEl) {
+          return;
+        }
+
+        // Get the src url.
+        let src = this.getAttribute("src");
+
+        // If the src attribute is a selector, get the url from the asset item.
+        if (src.charAt(0) === "#") {
+          const assetEl = document.getElementById(src.substring(1));
+          src = assetEl.getAttribute("src");
+        }
+
+        // Load the gltf model from the cache if it exists.
+        const gltf = GLTFCache[src];
+
+        if (gltf) {
+          // Use a cloned copy of the cached model.
+          const clonedGltf = cloneGltf(gltf);
+          this.onLoad(clonedGltf);
+          return;
+        }
+
+        const finalizeLoad = () => {
+          AFRAME.ANode.prototype.load.call(this, () => {
+            // Check if entity was detached while it was waiting to load.
+            if (!this.parentEl) {
+              return;
+            }
+
+            this.updateComponents();
+            if (this.isScene || this.parentEl.isPlaying) {
+              this.play();
+            }
+          });
+        };
+
+        const inflate = (model, callback) => {
+          inflateEntities("", this, model);
+          this.querySelectorAll(":scope > template").forEach(attachTemplate);
+          setTimeout(callback, 0);
```

Comment:
Why does this have to happen on next tick?

Reply:
Added a comment.